lib/matplotlib/axis.py

```diff
@@ -786,8 +795,10 @@ def set_tick_params(self, which='major', reset=False, **kw):
             if reset:
                 d.clear()
             d.update(kwtrans)
+
         if reset:
             self.reset_ticks()
+        # Is this "else" correct? Shouldn't kwargs be applied after a reset?
```

Comment:
`reset_ticks` deletes all ticks; they'll get recreated with the updated parameters from the dictionary.